packages/yourloops/components/patient-list/patient-list.hook.tsx

```diff
@@ -0,0 +1,293 @@
+        headerName: t('patient'),
+        hideable: false,
+        flex: 1,
+        headerClassName: classes.mandatoryCellBorder,
+        cellClassName: classes.mandatoryCellBorder,
+        renderCell: (params: GridRenderCellParams<GridRowModel, PatientProfile>) => {
+          const { firstName, fullName, lastName, email } = params.value
+          return <Box data-email={email}>{getUserName(firstName, lastName, fullName)}</Box>
+        },
+        sortComparator: sortByUserName
+      },
+      {
+        field: PatientListColumns.System,
+        headerName: t('system')
+      },
+      {
+        field: PatientListColumns.TimeOutOfRange,
+        type: 'number',
+        headerName: t('time-out-of-range-target'),
+        description: t('time-out-of-range-target-tooltip'),
+        flex: 0.5,
+        sortable: false,
+        renderCell: (params: GridRenderCellParams<GridRowModel, Alarms>) => {
+          const alarms = params.value
+          return <AlarmPercentageCell value={alarms.timeSpentAwayFromTargetRate} isAlarmActive={alarms.timeSpentAwayFromTargetActive} />
+        }
+      },
+      {
+        field: PatientListColumns.SevereHypoglycemia,
+        type: 'number',
+        headerName: t('alert-hypoglycemic'),
+        description: t('hypoglycemia-tooltip'),
+        sortable: false,
+        flex: 0.5,
+        renderCell: (params: GridRenderCellParams<GridRowModel, Alarms>) => {
+          const alarms = params.value
+          return <AlarmPercentageCell value={alarms.frequencyOfSevereHypoglycemiaRate} isAlarmActive={alarms.frequencyOfSevereHypoglycemiaActive} />
+        }
+      },
+      {
+        field: PatientListColumns.DataNotTransferred,
+        type: 'number',
+        headerName: t('data-not-transferred'),
+        description: t('data-not-transferred-tooltip'),
+        sortable: false,
+        flex: 0.5,
+        renderCell: (params: GridRenderCellParams<GridRowModel, Alarms>) => {
+          const alarms = params.value
+          return <AlarmPercentageCell value={alarms.nonDataTransmissionRate} isAlarmActive={alarms.nonDataTransmissionActive} />
+        }
+      },
+      {
+        type: 'string',
+        field: PatientListColumns.LastDataUpdate,
+        headerName: t('last-data-update'),
+        flex: 0.8
+      },
+      {
+        type: 'boolean',
+        field: PatientListColumns.Messages,
+        headerName: '',
+        width: 55,
+        renderCell: (params: GridRenderCellParams<GridRowModel, boolean>) => <MessageCell hasNewMessages={params.value} />
+      },
+      {
+        type: 'actions',
+        field: PatientListColumns.Actions,
+        headerName: 'Actions',
+        renderCell: (params: GridRenderCellParams<GridRowModel, string>) => <ActionsCell patientId={params.value} onClickRemove={onClickRemovePatient} />
+      }
+    ]
+  }, [t, classes.mandatoryCellBorder, sortByUserName, selectedTab, getUserName, onClickRemovePatient])
+
+  const rows: GridRowsProp = useMemo(() => {
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❔ question:** ‏So this is not a representation JSX of the rows, simply the props? Maybe rename it to `rowsProps`?